index.d.ts

```diff
@@ -2,33 +2,29 @@ export { } from "./jsx";
 
 export as namespace skate;
 
-export interface HasProps<El extends Component> {
-  props: { [key: string]: skate.PropOptions<El, any>; };
-}
-
-export interface OnUpdatedCallback {
-  updatedCallback(previousProps: any): boolean | undefined | void;
-}
-
-export interface OnRenderCallback {
-  renderCallback(): any;
-}
-
-export interface OnRenderedCallback {
-  renderedCallback(): void;
-}
-
-export class Component extends HTMLElement {
-  // static readonly props: { [name: string]: PropAttr<Component, any> };
+export class Component extends HTMLElement implements ICustomElementV1, ISkateLivecycle {
+  static readonly props: { [nameOrSymbol: string]: PropOptions<El, any> };
   static readonly observedAttributes: string[];
 
   // Custom Elements v1
   connectedCallback(): void;
   disconnectedCallback(): void;
-  attributeChangedCallback(name: string, oldValue: any, newValue: any): void;
+  attributeChangedCallback(name: string, oldValue: null | string, newValue: null | string): void;
+  adoptedCallback?(): void;
 
-  // SkateJS
-  updated(prev: any): boolean;
+  // SkateJS Livecycle
+  updatedCallback(previousProps: { [nameOrSymbol: string]: any }): boolean;
```

Comment:
We shoud use `previousProps: any` instead of `{ [nameOrSymbol: string]: any }`.
If code safety does not improment, it is better to be easy to learn and easy to use.